WordPress/Classes/ViewRelated/NUX/LoginFields.swift

```diff
@@ -0,0 +1,82 @@
+import Foundation
+
+/// LoginFields is a state container for user textfield input on the login screens
+/// as well as other meta data regarding the nature of a login attempt.
+///
+@objc
+class LoginFields: NSObject {
+    // These fields store user input from text fields.
+
+    /// Stores the user's account identifier (either email address or username) that is
+    /// entered in the login flow. By convention, even if the user is logging in
+    /// via an email address this field should store that value.
+    var username = ""
+
+    /// The user's password.
+    var password = ""
+
+    /// The site address if logging in via the self-hosted flow.
+    var siteAddress = ""
```

Comment:
I was debating if `siteAddress` should be optional (and maybe `multifactorCode` and `emailAddress`), but I think this is something to leave for now. Just something to keep in mind for the future.

Reply:
I actually started to go with it as an optional, but we're already checking for `empty` in several places.  This seemed like a lighter touch.  Def something we can revisit.